### DISCUSSION TITLE
Kibana Primary Shard is not Active

### DISCUSSION THREAD

**Discussion by palitech100** [Security-Onion-Solutions/securityonion#6539] [Unsupported Versions] [answered]
Hello,

I ran an update this morning and the update completed with 2 errors. I have a standalone box, which was rebooted after update. I am not able to pull up Kibana, I get: 404 Page not Found.  I get many errors in /opt/so/log/kibana/kibana.log (primary shard is not active and OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT). 

When I run curl localhost:9200/_cat/indices and curl localhost:9200/_cat/shards I get: curl: (52) Empty reply from server

Can you please help me. 

Thank you!

Martina

Some error messages:

![image](https://user-images.githubusercontent.com/96071754/145851057-5c40b526-742d-44af-8cc9-b0214eba537f.png)

 /root/soup.log
![image](https://user-images.githubusercontent.com/96071754/145848357-5d9db18c-6db3-4a0a-ad87-aa4995bfab31.png)

![image](https://user-images.githubusercontent.com/96071754/145850332-65a964a5-3cc5-4057-89ee-e6e617c64a65.png)

![image](https://user-images.githubusercontent.com/96071754/145850397-7e228e0e-3f94-4a7d-9ed2-daa0466f400f.png)

/opt/so/log/kibana/kibana.log
 [[.kibana_7.15.2_reindex_temp][0]] containing [930] requests]\"},{\"type\":\"unavailable_shards_exception\",\"reason\":\"[.kibana_7.15.2_reindex_temp][0] primary shard is not active Timeout: [1m], request: [BulkShardRequest [[.kibana_7.15.2_reindex_temp][0]] containing [930] requests]\"},{\"type\":\"unavailable_shards_exception\",\"reason\":\"[.kibana_7.15.2_reindex_temp][0] primary shard is not active Timeout: [1m], request: [BulkShardRequest [[.kibana_7.15.2_reindex_temp][0]] containing [930] requests]\"}]\n    at /usr/share/kibana/src/core/server/saved_objects/migrationsv2/actions/bulk_overwrite_transformed_documents.js:107:13\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\n    at stateActionMachine (/usr/share/kibana/src/core/server/saved_objects/migrationsv2/state_action_machine.js:67:28)\n    at migrationStateActionMachine (/usr/share/kibana/src/core/server/saved_objects/migrationsv2/migrations_state_action_machine.js:92:24)\n    at async Promise.all (index 0)"},"log":{"level":"ERROR","logger":"savedobjects-service"},"process":{"pid":293}}

{"ecs":{"version":"1.9.0"},"@timestamp":"2021-12-13T16:15:04.165+00:00","message":"[.kibana_task_manager] Action failed with 'search_phase_execution_exception: '. Retrying attempt 15 in 64 seconds.","log":{"level":"ERROR","logger":"savedobjects-service"},"process":{"pid":293}}
{"ecs":{"version":"1.9.0"},"@timestamp":"2021-12-13T16:15:04.167+00:00","message":"[.kibana_task_manager] OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT -> OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT. took: 64042ms.","log":{"level":"INFO","logger":"savedobjects-service"},"process":{"pid":293}}
{"ecs":{"version":"1.9.0"},"@timestamp":"2021-12-13T16:15:05.170+00:00","message":"[.kibana] Action failed with 'search_phase_execution_exception: '. Retrying attempt 15 in 64 seconds.","log":{"level":"ERROR","logger":"savedobjects-service"},"process":{"pid":293}}
{"ecs":{"version":"1.9.0"},"@timestamp":"2021-12-13T16:15:05.171+00:00","message":"[.kibana] OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT -> OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT. took: 64053ms.","log":{"level":"INFO","logger":"savedobjects-service"},"process":{"pid":293}}
{"ecs":{"version":"1.9.0"},"@timestamp":"2021-12-13T16:16:08.223+00:00","message":"[.kibana_task_manager] OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT -> FATAL. took: 64058ms.","log":{"level":"INFO","logger":"savedobjects-service"},"process":{"pid":293}}
{"ecs":{"version":"1.9.0"},"@timestamp":"2021-12-13T16:16:08.224+00:00","message":"Unable to complete saved object migrations for the [.kibana_task_manager] index: Unable to complete the OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT step after 15 attempts, terminating.","error":{"message":"Unable to complete saved object migrations for the [.kibana_task_manager] index: Unable to complete the OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT step after 15 attempts, terminating.","type":"Error","stack_trace":"Error: Unable to complete saved object migrations for the [.kibana_task_manager] index: Unable to complete the OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT step after 15 attempts, terminating.\n    at migrationStateActionMachine (/usr/share/kibana/src/core/server/saved_objects/migrationsv2/migrations_state_action_machine.js:144:29)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\n    at async Promise.all (index 1)\n    at SavedObjectsService.start (/usr/share/kibana/src/core/server/saved_objects/saved_objects_service.js:164:7)\n    at Server.start (/usr/share/kibana/src/core/server/server.js:352:31)\n    at Root.start (/usr/share/kibana/src/core/server/root/index.js:65:14)\n    at bootstrap (/usr/share/kibana/src/core/server/bootstrap.js:120:5)\n    at Command.<anonymous> (/usr/share/kibana/src/cli/serve/serve.js:226:5)"},"log":{"level":"FATAL","logger":"root"},"process":{"pid":293}}
{"ecs":{"version":"1.9.0"},"@timestamp":"2021-12-13T16:16:08.251+00:00","message":"Stopping all plugins.","log":{"level":"INFO","logger":"plugins-system.standard"},"process":{"pid":293}}
{"ecs":{"version":"1.9.0"},"@timestamp":"2021-12-13T16:16:08.253+00:00","message":"Monitoring stats collection is stopped","log":{"level":"INFO","logger":"plugins.monitoring.monitoring.kibana-monitoring"},"process":{"pid":293}}
{"ecs":{"version":"1.9.0"},"@timestamp":"2021-12-13T16:16:09.209+00:00","message":"[.kibana] OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT -> FATAL. took: 64039ms.","log":{"level":"INFO","logger":"savedobjects-service"},"process":{"pid":293}}
{"ecs":{"version":"1.9.0"},"@timestamp":"2021-12-13T16:16:38.260+00:00","message":"\"eventLog\" plugin didn't stop in 30sec., move on to the next.","log":{"level":"WARN","logger":"plugins-system.standard"},"process":{"pid":293}}

**Comment by dougburks** [comment, marked as answer]
Any additional clues in the Elasticsearch log?

**Reply by palitech100** [reply]
Hi Doug,

Thank you so much for following up. I was actually able to resolve the issue. Is it possible this issue was caused by having too many older indices open? I changed the cur_close_days back to the default 30, let the cron job run to close older indices, and restarted Kibana. Everything seems to be working fine now.

Thank you.

Martina

**Reply by dougburks** [reply]
Yes, having too many open indices can cause issues. From https://docs.securityonion.net/en/2.3/elasticsearch.html#closing-indices:
> Elasticsearch indices are closed based on the close setting shown in the global pillar above. This setting configures Curator to close any index older than the value given. The more indices are open, the more heap is required. Having too many open indices can lead to performance issues. There are many factors that determine the number of days you can have in an open state, so this is a good setting to adjust specific to your environment.
